libs/depgraph/depgraph_api.go

```diff
@@ -0,0 +1,608 @@
+// Beware that items are stored inside the graph and their content should not change
+// in any other way than through the Graph APIs. It is recommended to implement the Item
+// interface with *value* receivers (or alternatively pass *copied* item values to the graph).
+type Item interface {
+	// Name should return a unique string identifier for the item instance.
+	// Name is not allowed to contain the forward slash character ("/").
+	// It is required for the name to be unique only within item instances of the
+	// same type (see Type()). A globally unique item identifier is therefore
+	// a combination of the item type and the item name.
+	Name() string
+	// Label is an optional alternative name that does not have to be unique.
+	// It is only used in the graph visualization as the label for the graph node
+	// that represents the item. If empty string is returned, Item.Name() is used
+	// for labeling instead.
+	Label() string
+	// Type should return the name of the item type.
+	// Type is not allowed to contain the forward slash character ("/").
+	// This is something like reflect.TypeOf(item).Name(), but potentially much more
+	// human-readable.
+	// For example, type could be "Linux bridge".
+	Type() string
+	// Equal compares this and the other item instance (of the same type and name)
+	// for equivalency. For the purposes of state reconciliation (see libs/reconciler),
+	// Equal determines if the current and the new intended state of an item is equal,
+	// or if a state transition is needed.
+	// Note that if two items are equal, their dependencies should be the same!
+	Equal(Item) bool
+	// External should return true for items which are not managed (created/modified/deleted)
+	// by the caller/owner. This could be used for items created by other management agents
+	// or to represent system notifications (e.g. interface link is up).
+	// For reconciliation, the presence of external items in the graph is used only for
+	// dependency purposes (e.g. create A only after another microservice created B).
+	External() bool
+	// String should return a human-readable description of the item instance.
+	// (e.g. a network interface configuration)
+	String() string
+	// Dependencies returns a list of all dependencies that have to be satisfied before
+	// the item can be created (i.e. dependencies in the returned list are AND-ed).
+	// Should be empty for external item (see Item.External()).
+	Dependencies() []Dependency
+}
+
+// Node of a dependency graph. Stores an instance of the Item alongside
+// some state information, useful for reconciliation purposes.
+// It is up to the graph user to fill the content of the Node as needed
+// and then to call Graph.PutNode(node).
+// Note that Node implements methods ID() and AsGraph(). The latter allows
+// the node to present itself as a single-node read-only graph. Use it
+// only after the node was put into an actual graph with PutNode().
+type Node struct {
```

Comment:
Still confused by this. If this is for reconciliation, it probably should live in reconciler package. Is the root of a graph (or subgraph, same thing) an `Item` or `Node`? Why do I need a `Node` to wrap my `Item` to do things?

Reply:
I see that now. Will get rid of this Node wrapper and will respond to your other comment about nodes above.